### PR TITLE
feat(core): Expose the confirm password endpoint

### PR DESCRIPTION
**Pull request by provokateurin**
Second try of https://github.com/nextcloud/server/pull/43000

## Summary

This allows non-web clients to confirm the password in order to use endpoints that require it.

## Checklist

- Code is [properly formatted](https://docs.nextcloud.com/server/latest/developer_manual/digging_deeper/continuous_integration.html#linting)
- [Sign-off message](https://github.com/src-d/guide/blob/master/developer-community/fix-DCO.md) is added to all commits
- [ ] Tests ([unit](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#unit-tests), [integration](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#integration-tests), api and/or acceptance) are included
- [ ] Screenshots before/after for front-end changes
- [ ] Documentation ([manuals](https://github.com/nextcloud/documentation/) or wiki) has been updated or is not required
- [ ] [Backports requested](https://github.com/nextcloud/backportbot/#usage) where applicable (ex: critical bugfixes)

### CODE REVIEW

**Comment by provokateurin** [comment]
Ah of course this stills needs the frontend changes to work with the new endpoint.

**Review by nickvergessen — APPROVED** [review]
Fine by me, but I would prefer to add/duplicate it into https://github.com/nextcloud/server/blob/master/core/Controller/AppPasswordController.php to have it available as proper OCS API

**Comment by nickvergessen** [comment]
> Ah of course this stills needs the frontend changes to work with the new endpoint.

Currently this is non breaking and nothing needs changes? Or am I missing something?

**Review by bigcat88 — APPROVED** [review]
The only change is adding `NoCSRFRequired` - right?

**Comment by provokateurin** [comment]
You're both right, I thought I had converted it to OCS which is not true :sweat_smile:

File: core/Controller/AppPasswordController.php [hunk 1]
```diff
@@ -165,4 +172,35 @@ public function rotateAppPassword(): DataResponse {
...
 		]);
 	}
+
+	/**
+	 * Confirm the user password
+	 *
+	 * @NoAdminRequired
+	 * @BruteForceProtection(action=sudo)
+	 *
+	 * @license GNU AGPL version 3 or any later version
```
Comment (nickvergessen) [inline core/Controller/AppPasswordController.php:182]: Can be removed as it's the same as the rest of this file:

```suggestion
```

**Review by nickvergessen — APPROVED** [review]

**Review by Altahrim — APPROVED** [review]

**Pull request merged**